### ISSUE TITLE
Target/base branch without rebasing?

### ISSUE THREAD

**Issue by HenrikPoulsen** [peter-evans/create-pull-request#1085]
### Subject of the issue

Hello,
I have the situation where I want to automatically keep a git subtree up to date.
So in the happy path I just run it against the main branch of my repo, run the `git subtree pull` command and then the action creates a PR for it.
However there is another path here. And that is if the subtree has merge conflicts against your latest `main` branch, at which point it will fail and no PR can be created.
What I wanted to do in this scenario is to have a fallback step, where if there is a conflict, then instead I will `git checkout` an older revision (the first commit where the subtree was introduced), run the update on that (which will be without conflict) and then create a PR out of that and targetting the `main` branch.
This sadly does not work however, since it seems the action always wants to rebase. So when the `git subtree pull` works successfully, it then fails due to the rebasing getting that very same merge conflict, and thus no PR can be created.

Why am I doing this?
Because I would preferably not have to run `git subtree pull` myself, and rather want to get a github PR which has a merge conflict in it (as seen in picture)
![image](https://user-images.githubusercontent.com/472627/157451505-f8dfe39a-4cdc-4efb-91ea-711076284fc6.png)
So that I can either hit "Resolve conflicts" if its a trivial one, or just checkout that branch and then solve it that way.

Is there any way to get to this approach?
I did try creating a temporary branch which points at that old revision and pushing it. But in that scenario the PR created has the correct content, but it points at the temp branch as the target instead of `main`. So when I manually ediit the PR and set the target to `main` then I get the picture you saw above

### Steps to reproduce

Hopefully not needed

**Comment by peter-evans** [comment]
Hi @HenrikPoulsen 

> I did try creating a temporary branch which points at that old revision

This feels like the correct solution to me. The action needs branches to work correctly.

> the PR created has the correct content, but it points at the temp branch as the target instead of `main`

Set the `base` input to `main` and the action will rebase the changes made on the temporary branch on to `main` as the target branch.

**Comment by HenrikPoulsen** [comment]
> Set the base input to main and the action will rebase the changes made on the temporary branch on to main as the target branch.


That is what I did, but that doesn't work.
Because then the action triggers a merge conflict and aborts.
```
Rebasing commits made to branch 'subtree/auto-update-fallback-branch' on to base branch 'main'
...
CONFLICT (file location): lib/workers/repository/update/branch/lock-files/__snapshots__/index.spec.ts.snap added in 201aec8bd (Squashed 'packages/renovate/' changes from 426dcd20b..f337d50af) inside a directory that was renamed in HEAD, suggesting it should perhaps be moved to packages/renovate/lib/workers/repository/update/branch/lock-files/__snapshots__/index.spec.ts.snap.
```
It conflicts on all the git subtree changes there, because as part of how subtrees works it first creates a commit which is the upstream repo entirely as is, then it creates a merge commit which moves all those files to their permanent location.
So rebasing would always cause these issues when using git subtrees (as far as I can imagine anyway).

I would like the PR to be created targetting main, without rebasing. Since that would give a PR with that merge conflict visible (plus that git subtrees do need their merge commits)
Basically would like to be able to do something like
```
- uses: actions/create-pull-request@v3.14.0
  with:
    base: main
    rebase: false
```

But I assume that is it not feasible to have it not rebase?

**Comment by HenrikPoulsen** [comment]
Ah thanks for the tip. That works quite well. I didn't even know about github-script, so that will be very useful going forward.
I will go ahead and close this issue.
Thanks again
